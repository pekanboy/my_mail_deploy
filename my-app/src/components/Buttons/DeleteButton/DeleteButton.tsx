import React, {CSSProperties} from 'react';
import deleteImg from '../../../static/delete.svg';
import {Button} from '../Button/Button';

export interface DeleteButtonProps {
    clickHandler: () => void
}

export const DeleteButton: React.FC<DeleteButtonProps> = ({clickHandler}) => {
    const styles: CSSProperties = {
        borderLeftWidth: 0,
        borderRadius: '0 4px 4px 0',
    };

    return (
        <div>
            <Button
                avatar={deleteImg}
                clickHandler={clickHandler}
                text={'Delete'}
                styles={styles}
            />
        </div>
    );
};
