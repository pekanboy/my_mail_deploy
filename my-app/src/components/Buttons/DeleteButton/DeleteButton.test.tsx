import React from 'react';
import {configure, shallow, ShallowWrapper} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {DeleteButton, DeleteButtonProps} from './DeleteButton';

configure({adapter: new Adapter()});

const testData: DeleteButtonProps = {
    clickHandler: () => undefined,
};


describe('Test delete button component', () => {
    it('to match snapshot ', function() {
        const wrapper: ShallowWrapper = shallow(<DeleteButton {...testData} />);

        expect(wrapper).toMatchSnapshot();
    });
});
