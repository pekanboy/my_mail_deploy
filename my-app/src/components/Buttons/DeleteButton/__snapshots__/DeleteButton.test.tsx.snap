// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test delete button component to match snapshot  1`] = `
<div>
  <Button
    avatar="delete.svg"
    clickHandler={[Function]}
    styles={
      Object {
        "borderLeftWidth": 0,
        "borderRadius": "0 4px 4px 0",
      }
    }
    text="Delete"
  />
</div>
`;
